File: typescript/compete/code_golf/temperature-code-golf.ts
let c=Number(readline()),ts=readline().split(' ').map(i=>Number(i)),n,p
ts.map(t=>((n||-273)<=t&&t<0)?n=t:0<t&&t<=(p||5526)?p=t:t==0?n=p=0:0)
let b=(n||0)+(p||0),r=c==0?0:!n||!p?b:b<=0?p:n
console.log(r)

File: python3/compete/code_golf/temperature-code-golf.py
a=int(input())
b=list(map(int,input().split()))
c=[t for t in b if t<0]
d=[t for t in b if t>0]
if 0 in b:e=f=0
else:
 e=max(c)if any(c)else None
 f=min(d)if any(d)else None
if a==0:g=0
elif not any(c):g=f
elif not any(d):g=e
else:g=f if e+f<=0 else e
print(g)
